### PR TITLE
Fix NegawattFormatterMetersTotal to respond only to 'meters' request

### PR DESCRIPTION
**Pull request by shahardo**
#461

### CODE REVIEW

**Comment by amitaibu** [comment]
Use REStful's ``$this->valueMetadata``, and in the formatter ``$this->handler->getValueMetaData()``

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 1]
```diff
@@ -69,9 +69,194 @@ public function publicFieldsInfo() {
...
+      $result_type = 'meter';
+      $query->condition('e.meter_nid', $filter['meter']);
+      $query->fields('e', array('meter_nid'));
+      $query->groupBy('e.meter_nid');
+      unset($filter['meter']);
+    }
+
+    // Make sure we handled all the filter fields.
+    if (!empty($filter)) {
+      throw new \RestfulBadRequestException('Unknown fields in filter: ' . implode(', ', array_keys($filter)));
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: ``format_string()``

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 2]
```diff
@@ -69,9 +69,194 @@ public function publicFieldsInfo() {
...
+        }
+      }
+    }
+
+    // Prepare total section to output by the formatter.
+    $summary['type'] = $result_type;
+    $summary['values'] = $total;
+
+    // Pass info to the formatter
+    $this->valueMetadata[$_SERVER['REQUEST_TIME_FLOAT']]['summary'] = $summary;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: $_SERVER['REQUEST_TIME_FLOAT'] => resource name

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/NegawattEntityMeterBase.php [hunk 3]
```diff
@@ -131,4 +131,76 @@ protected function meterCategories($value) {
...
+   * {@inheritdoc}
+   *
+   * Prepare summary section for the formatter.
+   */
+  public function getQueryForList() {
+    // Handle summary section
+    // Pass to the formatter a summary of min and max electricity_time_interval.
+
+    // Prepare summary data for the formatter.
+    \NegawattEntityMeterBase::prepareSummary($this);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/NegawattEntityMeterBase.php]: Use ``public function viewMultiple(array $ids) {`` instead

**Comment by ceoaliongroo** [comment]
@shahardo is this issue wip?

**Comment by shahardo** [comment]
Yes. I'm working on it now
בתאריך 28 באפר' 2015 13:36,‏ "Carlos Mantilla" <notifications@github.com>
כתב:

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 4]
```diff
@@ -70,8 +70,271 @@ public function publicFieldsInfo() {
...
+    // Get query filter.
+    $request = $this->getRequest();
+    $filter = !empty($request['filter']) ? $request['filter'] : array();
+
+    if (array_key_exists('meter', $filter)) {
+      return $public_fields;
+    }
+
+    // No filter for meters, remove meters field from public-fields.
+    unset($public_fields['meter']);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: use ``access_callbacks`` on the meter

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 5]
```diff
@@ -70,8 +70,271 @@ public function publicFieldsInfo() {
...
+
+  /**
+   * Figure out vocab id from group id.
+   *
+   * The reverse of og_vocab_relation_get() function.
+   *
+   * @param $group_id
+   * @return mixed
+   */
+  protected function vocabIdFromGroupId($group_id) {
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: move to ``negwatt_``

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 6]
```diff
@@ -70,8 +70,271 @@ public function publicFieldsInfo() {
...
+  protected function prepareSummary() {
+    $request = $this->getRequest();
+    $filter = !empty($request['filter']) ? $request['filter'] : array();
+
+    // Load query and apply filter options.
+    $query = parent::getQuery();
+    $this->queryForListFilter($query);
+
+    // Add a query for meter_category.
+    $field = field_info_field(OG_VOCAB_FIELD);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: move to helper method

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/NegawattEntityMeterBase.php [hunk 7]
```diff
@@ -131,4 +131,79 @@ protected function meterCategories($value) {
...
+      // Add condition - the OG membership of the meter-node is equal to the
+      // account id in the request.
+      $query->join('field_data_field_contract_id', 'c', 'c.entity_id = e.meter_nid');
+      $query->condition('c.field_contract_id_value', $filter['contract']);
+      unset($filter['contract']);
+    }
+
+    // Make sure we handled all the filter fields.
+    if (!empty($filter)) {
+      throw new \Exception('Unknown fields in filter: ' . implode(', ', array_keys($filter)));
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/NegawattEntityMeterBase.php]: ``format_string()``

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/1.0/NegawattMeterResource.class.php [hunk 8]
```diff
@@ -60,4 +60,78 @@ public function publicFieldsInfo() {
...
+    // Prepare a min/max query.
+    $request = $this->getRequest();
+    $filter = empty($request['filter']) ? array() : $request['filter'];
+    unset($filter['has_electricity']);
+
+    $query = db_select('negawatt_electricity_normalized', 'e');
+
+    // Make sure we count only nodes of the current bundle.
+    $query->join('node', 'n', 'n.nid = e.meter_nid');
+//    $query->condition('n.type', $this->bundle);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/1.0/NegawattMeterResource.class.php]: remove

File: negawatt/modules/custom/negawatt_restful/negawatt_restful.module [hunk 9]
```diff
@@ -14,3 +14,21 @@ function negawatt_restful_ctools_plugin_directory($module, $plugin) {
...
+
+/**
+ * Figure out vocab id from group id.
+ *
+ * The reverse of og_vocab_relation_get() function.
+ *
+ * @param (int) $group_id
+ * @return (int) vocabulary_id
+ */
+function vocab_id_from_group_id($group_id) {
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/negawatt_restful.module]: ``negawatt_restful_...``

**Pull request closed without merging**